doc/multi_asic/SONiC_multi_asic_hld.md

```diff
@@ -0,0 +1,937 @@
+
+To support a router/switch with multiple switch ASICs, a disaggregated design approach is used. 
+The main idea is for the control/data plane related SONiC docker containers to be disaggregated per switch ASIC. 
+
+The below sonic dockers are instantiated per each ASIC:
+
+- lldp, teamd, bgp, syncd, swss, database
+
+- Each instance is responsible for the programming of a single ASIC.
+- Each ASIC is programmed and forwarding traffic independently
+- Each ASIC has its own SAI library and SDK instance.
+
+Below dockers are created for host and remain as single instance only:
+
+  snmp, telemetry, pmon
+
+- A separate database instance per ASIC
+- Configuration is generated and stored per ASIC.
+
+There is an additional database and lldp instance are running on the host
+
+
+The ability to program and forward traffic independently on each ASIC also depends on the HW architecture of the ASIC.  Some ASICs used in a chassis or distributed system may not support completely independent programming/forwarding when traffic from one port to another spans multiple ASICs.
+In this document, we will focus on a multi-ASIC system where ASICs can support completely independent programming/forwarding.  What this means is that traffic entering and leaving the ASIC will be in standard IP packet format, even within the router.  The ingress/egress pipeline to forward a packet is not separated into different ASICs and so there is no dependency of data-plane information shared among different ASICs in order to forward a packet from one port to another. 
+
+A linux network namespace is created for every ASIC that's controlled by the same CPU.
+This creates a separate linux network stack including routes and network devices for every ASIC.
+The interfaces for a given ASIC is linked to its corresponding namespace.
+
+In a multi-ASIC system, very commonly the ASICs are physically connected in a clos fabric topology. With sonic container dockers running as separate namespaces for each ASIC in a multi-ASIC system, we can model and configure the system as if there is a spine-leaf network topology within the box.

```

Comment:
How to configure multi-asic ? does it have a CLI? how does it configure different ASICs